### PR TITLE
tf-modules for lambda

### PR DESCRIPTION
**Pull request by somashekhar-moe**
1. tf-core module for lambda 
2. Lambda function name is kept it to be coming from variable rather than resource-identifies which might not make sense in case of lambda

### CODE REVIEW

File: lambda/lambda.tf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  dynamic "vpc_config" {
+    for_each = var.vpc_subnet_ids != null && var.vpc_security_group_ids != null ? [true] : []
+    content {
+      security_group_ids = var.vpc_security_group_ids
+      subnet_ids         = var.vpc_subnet_ids
+    }
+  }
+}
+
+resource "aws_cloudwatch_log_group" "example" {
```
Comment (sudip-moengage) [inline lambda/lambda.tf:33]: Change name from example to default

File: lambda/lambda.tf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  name              = "/aws/lambda/${var.function_name}"
+  retention_in_days = 7
```
Comment (sudip-moengage) [inline lambda/lambda.tf:35]: This sjoulfbe variable, which defaults to 7

File: lambda/locals.tf [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    ManagedBy   = "terraform"
+    Region      = data.aws_region.current.name
+    CreatedBy   = lower(var.created_by)
+    Environment = lower(var.environment)
+    Business    = lower(var.business_name)
+    Service     = lower(var.service_name)
+    SubService  = lower(var.subservice_name)
+  }
+
+  _resource_identifier = "${var.service_name}-${var.subservice_name}"
```
Comment (sudip-moengage) [inline lambda/locals.tf:12]: Add business as prefix as well, look at other modules for reference

**Review by sudip-moengage — CHANGES_REQUESTED** [review]
Where's cloudwatch log group being attached to lambda?

File: lambda/lambda.tf (hunk 1 above)
Comment (somashekhar-moe) [reply lambda/lambda.tf]: fixed

File: lambda/lambda.tf (hunk 2 above)
Comment (somashekhar-moe) [reply lambda/lambda.tf]: fixed

**Comment by somashekhar-moe** [comment]
> Where's cloudwatch log group being attached to lambda?

  name              = "/aws/lambda/${var.function_name}"
this is take care like this

File: lambda/lambda.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  }
+}
+
+resource "aws_cloudwatch_log_group" "default" {
+  name              = "/aws/lambda/${var.function_name}"
+  retention_in_days = var.retention_in_days
+}
+
+
```
Comment (sudip-moengage) [inline lambda/lambda.tf:37]: So many empty lines, remove this.

File: lambda/versions.tf [hunk 5]
```diff
@@ -0,0 +1,8 @@
+terraform {
```
Comment (sudip-moengage) [inline lambda/versions.tf:1]: I guess we can remove this file.

**Review by sudip-moengage — CHANGES_REQUESTED** [review]

**Pull request merged**